packages/ember-application/lib/system/application.js

```diff
@@ -671,20 +688,24 @@ var Application = Namespace.extend(DeferredMixin, {
     @method didBecomeReady
   */
   didBecomeReady: function() {
-    if (environment.hasDOM) {
-      this.__instance__.setupEventDispatcher();
-    }
+    if (true) {
+      if (environment.hasDOM) {
+        this.__legacyInstance__.setupEventDispatcher();
+      }
 
-    this.ready(); // user hook
-    this.startRouting();
+      this.ready(); // user hook
+      this.legacyStartRouting(this.__legacyInstance__);
 
-    if (!Ember.testing) {
-      // Eagerly name all classes that are already loaded
-      Ember.Namespace.processAll();
-      Ember.BOOTED = true;
-    }
+      if (!Ember.testing) {
+        // Eagerly name all classes that are already loaded
+        Ember.Namespace.processAll();
+        Ember.BOOTED = true;
+      }
 
-    this.resolve(this);
+      this.resolve(this);
```

Comment:
No, this is for the (deprecated) case of `Application.create().then(() => { })`. I don't think we can remove this until 2.0. Do you disagree?